tools/swift-stdlib-tool/swift-stdlib-tool.mm

```diff
@@ -976,22 +975,24 @@ int main(int argc, const char *argv[])
         }
 
         // Fix up src_dir and platform values.
-        if (!src_dir && !platform) {
-            // Neither src_dir nor platform is set. Die.
+        if (![src_dirs count] && !platform) {
+            // Neither src_dirs nor platform is set. Die.
             fail_usage("At least one of --source-libraries and --platform "
                        "must be set.");
         }
-        else if (!src_dir) {
-            // platform is set but src_dir is not. 
-            // Use platform to set src_dir relative to us.
-            src_dir = [[[self_executable stringByDeletingLastPathComponent]
-                        stringByDeletingLastPathComponent]
-                       sst_stringByAppendingPathComponents:
-                       @[ @"lib", @"swift-5.0", platform ]];
+        else if (![src_dirs count]) {
+            // platform is set but src_dirs is not.
+            // Use platform to set src_dirs relative to us.
+            NSString *root_path = [[self_executable stringByDeletingLastPathComponent]
+                                    stringByDeletingLastPathComponent];
+            src_dirs = [@[
+                [root_path sst_stringByAppendingPathComponents: @[ @"lib", @"swift-5.0", platform ]],
+                [root_path sst_stringByAppendingPathComponents: @[ @"lib", @"swift-5.5", platform ]],
+            ] mutableCopy];
```

Comment:
Would it make sense to future-proof this to list the contents of `${root_path}/lib` and grab all the directories named `swift-*`? (Or matching `swift-\d+(\.\d+)+` might be more precise, if we need that degree of precision?)

Reply:
I guess at this point no one knows how many times this will end up being helpful in the future, but seems reasonable, pushed that as an update